source/simulator/assemblers/stokes.cc

```diff
@@ -0,0 +1,393 @@
+                if (rebuild_stokes_matrix)
+                  {
+                    scratch.grads_phi_u[i_stokes] = scratch.finite_element_values[introspection.extractors.velocities].symmetric_gradient(i,q);
+                    scratch.div_phi_u[i_stokes]   = scratch.finite_element_values[introspection.extractors.velocities].divergence (i, q);
+                  }
+                ++i_stokes;
+              }
+            ++i;
+          }
+
+
+        // Viscosity scalar
+        const double eta = (rebuild_stokes_matrix
+                            ?
+                            scratch.material_model_outputs.viscosities[q]
+                            :
+                            std::numeric_limits<double>::quiet_NaN());
+
+        const SymmetricTensor<4,dim> &stress_strain_director =
+          scratch.material_model_outputs.stress_strain_directors[q];
+        const bool use_tensor = (stress_strain_director !=  dealii::identity_tensor<dim> ());
+
+        const Tensor<1,dim>
+        gravity = this->get_gravity_model().gravity_vector (scratch.finite_element_values.quadrature_point(q));
+
+        const double density = scratch.material_model_outputs.densities[q];
+
+        const double JxW = scratch.finite_element_values.JxW(q);
+
+        for (unsigned int i=0; i<stokes_dofs_per_cell; ++i)
+          {
+            data.local_rhs(i) += (density * gravity * scratch.phi_u[i])
+                                 * JxW;
+
+            if (rebuild_stokes_matrix)
+              for (unsigned int j=0; j<stokes_dofs_per_cell; ++j)
+                {
+                  data.local_matrix(i,j) += ( (use_tensor ?
+                                               eta * 2.0 * (scratch.grads_phi_u[i] * stress_strain_director * scratch.grads_phi_u[j])
+                                               :
+                                               eta * 2.0 * (scratch.grads_phi_u[i] * scratch.grads_phi_u[j]))
+                                              // assemble \nabla p as -(p, div v):
+                                              - (pressure_scaling *
+                                                 scratch.div_phi_u[i] * scratch.phi_p[j])
+                                              // assemble the term -div(u) as -(div u, q).
+                                              // Note the negative sign to make this
+                                              // operator adjoint to the grad p term:
+                                              - (pressure_scaling *
+                                                 scratch.phi_p[i] * scratch.div_phi_u[j]))
+                                            * JxW;
+                }
+          }
+      }
+  }
+
+  template <int dim>
+  void
+  StokesAssembler<dim>::
+  local_assemble_stokes_compressible_diffusion (const double                                     /*pressure_scaling*/,
```

Comment:
Yes it would be better to add some clarification to them. I was thinking about dropping the `local_assemble_stokes` part of the name, since all functions within `StokesAssembler` do that.